module/input/SensorFilter/src/SensorFilter.hpp

```diff
@@ -151,33 +167,183 @@ namespace module::input {
+        void configure_ukf(const Configuration& config);
+
+        /// @brief Configure Kalman filter
+        void configure_kf(const Configuration& config);
+
+        /// @brief Configure Mahony filter
+        void configure_mahony(const Configuration& config);
+
+        /// @brief Updates the sensors message with odometry data filtered using UKF. This includes the
+        // position, orientation, velocity and rotational velocity of the torso in world space.
+        /// @param sensors The sensors message to update
+        /// @param previous_sensors The previous sensors message
+        /// @param raw_sensors The raw sensor data
+        void update_odometry_ukf(std::unique_ptr<Sensors>& sensors,
+                                 const std::shared_ptr<const Sensors>& previous_sensors,
+                                 const RawSensors& raw_sensors);
+
+        /// @brief Updates the sensors message with odometry data filtered using Kalman Filter. This includes the
+        // position, orientation, velocity and rotational velocity of the torso in world space.
+        /// @param sensors The sensors message to update
+        /// @param previous_sensors The previous sensors message
+        /// @param raw_sensors The raw sensor data
+        void update_odometry_kf(std::unique_ptr<Sensors>& sensors,
+                                const std::shared_ptr<const Sensors>& previous_sensors,
+                                const RawSensors& raw_sensors);
+
+
+        /// @brief Updates the sensors message with odometry data filtered using MahonyFilter. This includes the
+        // position, orientation, velocity and rotational velocity of the torso in world space.
+        /// @param sensors The sensors message to update
+        /// @param previous_sensors The previous sensors message
+        /// @param raw_sensors The raw sensor data
+        void update_odometry_mahony(std::unique_ptr<Sensors>& sensors,
+                                    const std::shared_ptr<const Sensors>& previous_sensors,
+                                    const RawSensors& raw_sensors);
+
+        /// @brief Display debug information
+        /// @param sensors The sensors message to update
+        /// @param raw_sensors The raw sensor data
+        void debug_sensor_filter(std::unique_ptr<Sensors>& sensors, const RawSensors& raw_sensors);
 
     private:
-        // Current state of the button pushes
-        // used to debounce button presses
-        bool leftDown   = false;
-        bool middleDown = false;
+        /// @brief Dead reckoning yaw orientation of the robot in world space
+        double yaw = 0;
+
+        /// @brief Transform of torso from world space
+        Eigen::Isometry3d Hwt = Eigen::Isometry3d::Identity();
+
+        /// @brief Current walk command
+        Eigen::Vector3d walk_command = Eigen::Vector3d::Zero();
+
+        /// @brief Bool to indicate if the robot is falling
+        bool falling = false;
```

Comment:
is this needed?

Reply:
It is, forgot to implement though. We should try and replace these with Director stability stuff though later.